discord/ext/menus/__init__.py

```diff
@@ -242,21 +301,23 @@ async def on_thumbs_down(self, payload):
     emoji: Union[:class:`str`, :class:`discord.PartialEmoji`]
         The emoji to use for the button.
     """
+
     def decorator(func):
         func.__menu_button__ = _cast_emoji(emoji)
         func.__menu_button_kwargs__ = kwargs
         return func
+
     return decorator
 
+
 class _MenuMeta(type):
-    @classmethod
-    def __prepare__(cls, name, bases, **kwargs):
-        # This is needed to maintain member order for the buttons
-        return OrderedDict()
+    if TYPE_CHECKING:
+        __menu_buttons__: List[_BtnFunc]
 
-    def __new__(cls, name, bases, attrs, **kwargs):
+    def __new__(cls: Type[type], name: str, bases: Tuple[type, ...], attrs: Dict[str, Any], **kwargs: Any):
         buttons = []
-        new_cls = super().__new__(cls, name, bases, attrs)
+
+        new_cls = cast(Type[_MenuMeta], super().__new__(cls, name, bases, attrs))
```

Comment:
type.__new__ doesn't return ``type[type]``
```suggestion
        new_cls = cast(_MenuMeta, super().__new__(cls, name, bases, attrs))
```